1- Suppose you invested in Bitcoin at the end of 2020 when Bitcoin gained a lot of value. What would be your money at the end of a week if you had invested $1000 with an average daily increase of 12% ? You can solve the problem using Python.


Help

Create a variable capital ($1000)

Create a variable for daily growth (12%)

Create a variable for period (7)

Calculate the final growth rate

Calculate result

Print result

In [7]:
capital = 1000
daily_rate = 0.12
period = 7

result = capital * ((1 + daily_rate)**7)
gain = result - capital
print(f"Amount of money at the end: {gain:.2f}")


Amount of money at the end: 1210.68


2- Print the text in quotes with Python. However, you must get the numbers from variables using .format() notation. Because the text is long, you might consider writing in two lines:

"When we buy bitcoin with 1000 USD at the beginning of the week, we would earn 1210.68 USD at the end of the week, with an average gain of 12\%."

In [6]:
print("""When we buy bitcoin with {} USD at the beginning of the week, 
      we would earn {:.2f} USD at the end of the week, with an average gain of {:.0f}%.""".format(capital, gain, daily_rate*100))


When we buy bitcoin with 1000 USD at the beginning of the week, 
      we would earn 1210.68 USD at the end of the week, with an average gain of 12%.


3- Get the temperature in Fahrenheit from user and write a code to convert it to Celcius. For conversion, you can use this formula: C = (5/9) * (F - 32)
Enter the temperature in Fahrenheit:

user --> 26

output --> Temperature (C) : -3.33

In [26]:
F = int(input("Enter the temperature in Fahrenheit: "))
C = (5/9) * (F - 32)
print(f"{F} in Fahrenheit is {C:.2f} in Celcius")


26 in Fahrenheit is -3.33 in Celcius


4- Get a three digit number the from user and calculate the sum of the digits in the integer.
user --> 365

output --> "The sum of digits in the number is 14

In [33]:
num = input("Enter a three digit number: ")
total = int(num[0]) + int(num[1]) + int(num[2])
print(f"The sum of digits in the number is: {total}")

The sum of digits in the number is: 14


In [34]:
#Alternative method
num = input("Enter a three digit number: ")
total = 0
for i in num:
    total += int(i)
print(f"The sum of digits in the number is: {total}")

The sum of digits in the number is: 14


5- Write some code to calculate the hypotenuse of a right angled triangle. Get the side lengths from the user.
user --> first side lenth : 6

user --> first side lenth : 8

output --> "The length of the hypotenuse is 10

In [2]:
side1 = int(input("first side lenth : "))
side2 = int(input("Second side lenth : "))
hypotenuse = (side1**2 + side2**2) ** (1/2)
print(f"The length of the hypotenuse is {round(hypotenuse)}")

The length of the hypotenuse is 10
